### PR TITLE
Docs: misc updates

### PR DESCRIPTION
**Pull request by hirooih**
- ctags(1): group options by categories
  - reorder "New and extended options" section in news.rst in the same ordering
- move some sections in news.rst to proper chapters
- create a chapter "Option files" extracted from "optlib" chapter and update the contents
- reorder chapters again.
- add new "writing a parser in C" section
- etc.

See commit logs for more details.

### CODE REVIEW

**Comment by hirooih** [comment]
The fail on Centos

```
#!/bin/bash -eo pipefail
yum install -y git gcc make autoconf automake pkgconfig
Loaded plugins: fastestmirror, ovl
Setting up Install Process
http://vault.centos.org/centos/6/os/x86_64/repodata/repomd.xml: [Errno 14] PYCURL ERROR 7 - "Failed to connect to 2600:1f13:2e7:2901:d2b0:c3b2:9347:934c: Cannot assign requested address"
Trying other mirror.
Error: Cannot retrieve repository metadata (repomd.xml) for repository: base. Please verify its path and try again

Exited with code exit status 1
CircleCI received exit code 1
```

I have no idea about this.

**Comment by masatake** [comment]
#2811 may solve the issue.

**Comment by masatake** [comment]
Could you rebase on the latest master?

**Comment by hirooih** [comment]
> Could you rebase on the latest master?

Now passed.  Thanks.

File: docs/contributions.rst [hunk 1]
```diff
@@ -251,32 +253,53 @@ Write a test case for Tmain or Units.
...
 
-Add your `.c` file to `source.mak`.
+Note: ``makefiles/translator_input.mak`` and ``main/parsers.h`` in this commit
+were renamed to ``makefiles/optlib2c_input.mak`` and ``main/parser_p.h``
+respectively in the latest Universal Ctags distribution.
 
-In addition, update `win32/ctags_vs2013.vcxproj` and
-`win32/ctags_vs2013.vcxproj.filters`. Otherwise our CI process run on
-Appveyor will fail.
+.. TODO: Is this correct?  Is there pull-request using the current files?
```
Comment (masatake) [inline docs/contributions.rst:302]: See #2765.

Comment (hirooih) [reply docs/contributions.rst]: Thanks.

I've changed as;

```
See `this pull request <https://github.com/universal-ctags/ctags/pull/2765>`_
for the `Meson` parser as an example of optlib parser.
```

**Review by masatake — APPROVED** [review]

**Comment by hirooih** [comment]
Thank you for your review.

**Pull request merged**